### PR TITLE
Global contexts restructurization

### PR DESCRIPTION
**Pull request by lukachi**

### CODE REVIEW

**Comment by lukachi** [comment]
Seems like every Web3 or Auth or Snap, ...etc contexts in `main.ts` could rerender router provider

File: src/contexts/auth.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const [isJwtValid, setIsJwtValid] = useState(false)
+
+  const isAuthorized = useMemo(
+    () => (provider?.isConnected && isSnapInstalled && isJwtValid) || false,
+    [isJwtValid, isSnapInstalled, provider?.isConnected],
+  )
+
+  const _setJwt = useCallback((jwt: string) => {
+    authStore.setJwt(jwt)
+  }, [])
```
Comment (ardier16) [inline src/contexts/auth.tsx:34]: IMO we don't need this wrapper and useCallback at all

we can use `authStore.setJwt` directly as it's defined outside of the React scope

File: src/contexts/auth.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+  const checkJwtValid = useCallback(async () => {
+    //Todo: add real logic
```
Comment (ardier16) [inline src/contexts/auth.tsx:37]:
```suggestion
    // TODO: add implementation
```

File: src/routes.tsx [hunk 3]
```diff
@@ -44,19 +41,58 @@ export const AppRoutes = () => {
...
+    checkSnapExists,
+    connectOrInstallSnap,
+    authorize,
+  ])
+
+  const { data: isAppInitialized } = useLoading(false, initRouter, {
+    loadOnMount: true,
+  })
+
+  if (!isAppInitialized) return null
```
Comment (ardier16) [inline src/routes.tsx:85]: we displayed a loader before

File: src/pages/SignIn.tsx [hunk 4]
```diff
@@ -20,6 +27,8 @@ export default function SignIn() {
 
     try {
       await login()
+
+      // navigate(RoutePaths.Orgs)
```
Comment (ardier16) [inline src/pages/SignIn.tsx:31]: remove comments

**Review by ardier16 — CHANGES_REQUESTED** [review]

**Review by ardier16 — APPROVED** [review]

**Pull request merged**